test/actions/katello/content_view_puppet_environment_test.rb

```diff
@@ -96,6 +96,23 @@ class CloneToEnvironmentTest < TestBase
     end
   end
 
+  class CopyContentsTest < TestBase
+    let(:action_class) { ::Actions::Pulp::ContentViewPuppetEnvironment::CopyContents }
+    let(:action) { create_action action_class }
+
+    it 'plans and can invoke' do
+      target = puppet_env
+      source = katello_repositories(:p_forge)
+      puppet_module = source.puppet_modules.first
+
+      plan_action action, target, :source_repository_id => source.id, :puppet_modules => [puppet_module]
+      Katello::Pulp::Repository::Puppet.any_instance.expects(:copy_contents).with do |repo, options|
+        repo.pulp_id == target.pulp_id && options[:puppet_modules].to_a == [puppet_module]
```

Comment:
What is this testing? that the pulp_id is populated with the target ID if the puppet  module is in `options`?

Reply:
its checking that what is passed in is what is expected, it similar to  .expects().with(arg1, arg2), but in this case we can't actually do an equality check on the entire object, so instead we check the pulp_id to make sure its what is expected.